### DISCUSSION TITLE
Info yaml BMS Seplos MQTT Wemos D1

### DISCUSSION THREAD

**Discussion by saviothecnic** [syssi/esphome-seplos-bms#70] [General]
Friends I have connected as in the photo a
Wemos D1 and a RS-485 TTL converter

I have a Seplos BMS 10E
Can you tell me or attach which yaml I need to load to have everyone
Seplos BMS sensors visible in MQTT

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/d6b0afd1-6a7a-43dd-a125-949ca06edd6e)

**Comment by saviothecnic** [comment]
My yaml

```
substitutions:
  name: seplos-bms
  device_description: "Monitor a Seplos BMS via RS485"
  external_components_source: github://syssi/esphome-seplos-bms@main
  tx_pin: GPIO1
  rx_pin: GPIO3

esphome:
  name: ${name}
  comment: ${device_description}
  project:
    name: "syssi.esphome-seplos-bms"
    version: 1.1.0

esp8266:
  board: d1_mini

external_components:
  - source: ${external_components_source}
    refresh: 0s

web_server:
  port: 80

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

ota:

logger:
  level: DEBUG

# If you use Home Assistant please remove this `mqtt` section and uncomment the `api` component!
# The native API has many advantages over MQTT: https://esphome.io/components/api.html#advantages-over-mqtt
mqtt:
  broker: !secret mqtt_host
  username: !secret mqtt_username
  password: !secret mqtt_password
  id: mqtt_client

# api:

uart:
  id: uart_0
  # Please set the default baudrate of your Seplos BMS model here. It's sometimes 19200 baud instead of 9600.
  baud_rate: 9600
  tx_pin: ${tx_pin}
  rx_pin: ${rx_pin}
  # The increased RX buffer size is important because
  # the full BMS response must fit into the buffer
  rx_buffer_size: 384

seplos_modbus:
  id: modbus0
  uart_id: uart_0
  rx_timeout: 150ms

seplos_bms:
  id: bms0
  # Dip switch configuration of a single pack setup / address 0x0
  #  8    7    6    5    4    3    2    1
  # off, off, off, off, off, off, off, off
  address: 0x00
  # Known protocol versions: 0x20 (Seplos), 0x26 (Boqiang)
  protocol_version: 0x20
  seplos_modbus_id: modbus0
  update_interval: 10s

sensor:
  - platform: seplos_bms
    min_cell_voltage:
      name: "${name} min cell voltage"
    max_cell_voltage:
      name: "${name} max cell voltage"
    min_voltage_cell:
      name: "${name} min voltage cell"
    max_voltage_cell:
      name: "${name} max voltage cell"
    delta_cell_voltage:
      name: "${name} delta cell voltage"
    average_cell_voltage:
      name: "${name} average cell voltage"
    cell_voltage_1:
      name: "${name} cell voltage 1"
    cell_voltage_2:
      name: "${name} cell voltage 2"
    cell_voltage_3:
      name: "${name} cell voltage 3"
    cell_voltage_4:
      name: "${name} cell voltage 4"
    cell_voltage_5:
      name: "${name} cell voltage 5"
    cell_voltage_6:
      name: "${name} cell voltage 6"
    cell_voltage_7:
      name: "${name} cell voltage 7"
    cell_voltage_8:
      name: "${name} cell voltage 8"
    cell_voltage_9:
      name: "${name} cell voltage 9"
    cell_voltage_10:
      name: "${name} cell voltage 10"
    cell_voltage_11:
      name: "${name} cell voltage 11"
    cell_voltage_12:
      name: "${name} cell voltage 12"
    cell_voltage_13:
      name: "${name} cell voltage 13"
    cell_voltage_14:
      name: "${name} cell voltage 14"
    cell_voltage_15:
      name: "${name} cell voltage 15"
    cell_voltage_16:
      name: "${name} cell voltage 16"
    temperature_1:
      name: "${name} temperature 1"
    temperature_2:
      name: "${name} temperature 2"
    temperature_3:
      name: "${name} temperature 3"
    temperature_4:
      name: "${name} temperature 4"
    temperature_5:
      name: "${name} environment temperature"
    temperature_6:
      name: "${name} mosfet temperature"
    total_voltage:
      name: "${name} total voltage"
    current:
      name: "${name} current"
    power:
      name: "${name} power"
    charging_power:
      name: "${name} charging power"
    discharging_power:
      name: "${name} discharging power"
    residual_capacity:
      name: "${name} residual capacity"
    battery_capacity:
      name: "${name} battery capacity"
    rated_capacity:
      name: "${name} rated capacity"
    state_of_charge:
      name: "${name} state of charge"
    charging_cycles:
      name: "${name} charging cycles"
    state_of_health:
      name: "${name} state of health"
    port_voltage:
      name: "${name} port voltage"
```

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/8ce8074e-9238-4eba-a46a-16d989664f0d)

No Work :(

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/38fd0793-3567-4d95-9569-3e26a6224294)

**Comment by saviothecnic** [comment]
I also tried with this other module but nothing

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/b6a52dc5-e735-4927-a5f6-4ab042386ed5)

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/2a46a86f-70d0-4938-8a18-c7c0bc6db9ce)

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/0b2b93b3-26de-41e7-8f2d-378b6b263e44)

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/af0a350a-ed31-4a8d-b027-93e9efc58ad8)

Where am I wrong

**Reply by syssi** [reply]
Does the TX LED blink periodically?

**Reply by saviothecnic** [reply]
only one of the two LEDs flashes very slowly
when in doubt I ordered another TTL RS-485 module
As soon as it arrives I'll try with the new module
To remove any doubts that my TTL RS-485 adapter is defective

**Reply by syssi** [reply]
It's important the LED labeled with `TXD` blinks periodically. If the RX LED is the only one which blinks your wiring isn't correct.

**Reply by saviothecnic** [reply]
Tutto ok qui video del test https://youtu.be/gc0NTUYBpK4

Con Modulo Atom

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/ae83b12b-bb3c-45f4-a598-9cbbc14dabc0)

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/1f3ba4f5-2150-46ce-9d11-9cca59dd1324)

**Reply by saviothecnic** [reply]
It seems to read all the data but I don't understand what these are

Invalid header

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/3974409b-9daa-4bce-8e70-200c9d8268e6)

**Reply by syssi** [reply]
Which GPIOs do you use for the serial communication? Is your solution ESP8266 or ESP32 based? I don't know the M5 products in detail.

**Reply by saviothecnic** [reply]
And based on ESP32
this is the module configuration
![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/978ba70c-cd2d-4b09-936f-f1e5dc55d58a)

**Reply by syssi** [reply]
Could you provide some more log messages in plaintext? Your "invalid header" issue is unusual.

**Reply by saviothecnic** [reply]
Of course, my friend is fully available to give you the information you need to understand

[logs_seplos-bat1_logs.txt](https://github.com/syssi/esphome-seplos-bms/files/13310925/logs_seplos-bat1_logs.txt)

this is the yamil remove.txt otherwise it wouldn't have been attached

[seplos-bat1.yaml.txt](https://github.com/syssi/esphome-seplos-bms/files/13310932/seplos-bat1.yaml.txt)

My BMS Seplos model

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/a44de749-acda-4805-b4cf-5c8138a955dd)

**Reply by saviothecnic** [reply]
@syssi Dude I added other battery packs in parallel
to my main Seplos
And I have to tell you that how do you add battery banks
and then you make them communicate the batteries with each other
the system stops working
Do you think you can solve this thing?

**Reply by syssi** [reply]
Could you post your config YAML and describe your setup / dip switches of the battery banks? Could you provide some logs too?

**Comment by FritsOV** [comment]
baud rate of the 10E is 19200 
not that will help. I got the same problem with a Live minikit Esp32. One led blinks tx does not come up. 
Changing A and B orange and orange/white does not make a difference... 

Have 4 packs connected with the RS port and Victrons system on the Can port. 

I am out of clues also

**Reply by saviothecnic** [reply]
@syssi 
I am connected in RS-485 with Voltronic at 9600
but anyway as soon as I added a parallel bench
the system is finished working :(
I hope they fix this
Because the system was nice and I was happy to use it

My DIP SW

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/7836ed50-d381-4144-bcd6-6255d26ebc5a)

**Reply by saviothecnic** [reply]
What's new guys?
If you need more proof, just say so
but I think I have answered all the information you asked me

**Reply by syssi** [reply]
Could you rephrase this statement:

> I am connected in RS-485 with Voltronic at 9600
> but anyway as soon as I added a parallel bench
> the system is finished working :(
> I hope they fix this
> Because the system was nice and I was happy to use it

Has it worked in the past? What did you change? Does it work of no other device is attached to the RS485 port?

**Reply by saviothecnic** [reply]
Yes, I installed it when I bought the first battery bank
with BMS Seplos V 2.0 and everything worked
The system stopped working when I started adding more battery banks and therefore more BMS in parallel
To my BMS Seplos V2

This works if I only have one BMS connected to the system
This works even if I only have one BMS
plus Inverter connected to the Seplos V2 BMS and ESP32 connected to the BMS

Only if I connect two or more BMS in parallel does the system no longer work

I currently have 4 bms in parallel
Maybe I just need to modify my yaml files to use it with 2 3 or 4 bms in parallel if this is the problem
Can you tell me how I have to modify the yaml file?
Thanks friend

**Reply by syssi** [reply]
Can you tell me all the devices connected to the bus?

**Reply by saviothecnic** [reply]
There are 4 battery banks with Seplos V2.0 BMS
In this photo the 4 is not yet mounted, now it is
In the first RS-485 I had connected ESP-32 in the last RS-485 I had connected Voltronik King UNO Inverter
I also tried without connecting the Volronic king uno
and connect as well as the BMS to each other
ESP-32 on the first or last battery bank without success
![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/2e789dbe-ec36-43f9-b4cc-70d43542ebb4)

**Reply by syssi** [reply]
Could you tell me the DIP switch setting of each battery bank?

**Reply by saviothecnic** [reply]
Dude this is how the DP Switches are set up
practically as the manual recommends

![image](https://github.com/syssi/esphome-seplos-bms/assets/6697807/b9bcde6d-b828-4acd-804f-ba2ae77791d7)

**Comment by zekeboz** [comment]
Were you able to resolve? Appears this maybe a hardware issue from what i read in another thread. 

Has anyone tried to poll the BLE connection, similar to this project: [https://github.com/fl4p/batmon-ha](https://github.com/fl4p/batmon-ha) ?
